packages/sessions-sdk-ts/src/index.ts

```diff
@@ -489,3 +498,102 @@ export type Session = {
   ) => Promise<TransactionResult>;
   sessionInfo: z.infer<typeof sessionInfoSchema>;
 };
+
+const TRANSFER_MESSAGE_HEADER = `Fogo Transfer:
+Signing this intent will transfer the tokens as described below.
+`;
+
+type SendTransferOptions = {
+  adapter: SessionAdapter;
+  walletPublicKey: PublicKey;
+  signMessage: (message: Uint8Array) => Promise<Uint8Array>;
+  mint: PublicKey;
+  amount: bigint;
+  recipient: PublicKey;
+};
+
+export const sendTransfer = async (options: SendTransferOptions) => {
+  const sourceAta = getAssociatedTokenAddressSync(
+    options.mint,
+    options.walletPublicKey,
+  );
+  const destinationAta = getAssociatedTokenAddressSync(
+    options.mint,
+    options.recipient,
+  );
+  const program = new IntentTransferProgram(
+    new AnchorProvider(options.adapter.connection, {} as Wallet, {}),
+  );
+  const umi = createUmi(options.adapter.connection.rpcEndpoint);
+  const metaplexMint = metaplexPublicKey(options.mint.toBase58());
+  const metadataAddress = findMetadataPda(umi, { mint: metaplexMint })[0];
+  const metadata = await safeFetchMetadata(umi, metadataAddress);
+  const symbol = metadata?.symbol ?? undefined;
+
+  return options.adapter.sendTransaction(undefined, [
+    createAssociatedTokenAccountIdempotentInstruction(
+      options.adapter.payer,
+      destinationAta,
+      options.recipient,
+      options.mint,
+    ),
+    await buildTransferIntentInstruction(program, options, symbol),
+    await program.methods
+      .sendTokens()
+      .accounts({
+        destination: destinationAta,
+        mint: options.mint,
+        source: sourceAta,
+        // eslint-disable-next-line unicorn/no-null
+        metadata: null,
+        sponsor: options.adapter.payer,
+        ...(symbol !== undefined && {
+          metadata: new PublicKey(metadataAddress),
+        }),
+      })
+      .instruction(),
+  ]);
+};
+
+const buildTransferIntentInstruction = async (
+  program: IntentTransferProgram,
+  options: SendTransferOptions,
+  symbol?: string,
+) => {
+  const [nonce, { decimals }] = await Promise.all([
+    getNonce(program, options.walletPublicKey),
+    getMint(options.adapter.connection, options.mint),
+  ]);
+  const message = new TextEncoder().encode(
+    [
+      TRANSFER_MESSAGE_HEADER,
+      serializeKV({
+        version: "0.1",
+        chain_id: options.adapter.chainId,
+        token: symbol ?? options.mint.toBase58(),
+        amount: amountToString(options.amount, decimals),
```

Comment:
it seems strange that the user provides this as a string then you convert to bigint and back to string here. Should you just use the string that the user has given you (after checking?)

Reply:
Yeah, I need to revisit this, the original idea was that `@fogo/session-sdk` (the package that's platform agnostic and can be used from any js code) would always take `bigint`, because there's not really a valid assumption that the value is even coming from user input.  However, that means everywhere we take `bigint`, we also have to resolve the mint to figure out the number of decimals and convert to string.

I'm starting to think it makes more sense to just take strings everywhere in the sdk.  I don't even think we need to check the string, it will fail in the contract if the string is invalid, and the React sdk has form validation to prevent passing an invalid value down.